sdk/internal/logger/logger.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package logger
+
+import (
+	"fmt"
+	"os"
+	"time"
+)
+
+// LogClassification is used to group entries.  Each group can be toggled on or off.
+type LogClassification string
+
+const (
+	// LogRequest entries contain information about HTTP requests.
+	// This includes information like the URL, query parameters, and headers.
+	LogRequest LogClassification = "Request"
+
+	// LogResponse entries contain information about HTTP responses.
+	// This includes information like the HTTP status code, headers, and request URL.
+	LogResponse LogClassification = "Response"
+
+	// LogRetryPolicy entries contain information specific to the retry policy in use.
+	LogRetryPolicy LogClassification = "RetryPolicy"
+
+	// LogLongRunningOperation entries contain information specific to long-running operations.
+	// This includes information like polling location, operation state and sleep intervals.
+	LogLongRunningOperation LogClassification = "LongRunningOperation"
+)
```

Comment:
Just the type declaration needs to stay be in the internal module, these should be in `azcore` so that customers can use them.